src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Errors/UserStringBuilder.cs

```diff
@@ -656,21 +656,20 @@ private void ErrAppendType(CType pType, SubstContext pctx, bool fArgs)
                             // Add [] with (rank-1) commas inside
                             ErrAppendChar('[');
 
-#if ! CSEE
```

Comment:
It appears to be pieces that supports Expression Evaluator that is used at debug time. They make no sense in the dynamic binder code.  
I guess they were just ported together with the rest of the code. Perhaps the initial pass used some automated translator and it picked up everything, or it was expected that there would be a need for a special binder - specifically for EE to call into, but that need never materialized...

Anyways, I am not aware of any CSEE flavor of the runtime binder.  
The "CSEE" seems just dead code and can/should be removed.